cni: Avoid returning error in DEL command

vadorovsky opened the pull request:
CNI DEL operations might be never retried by the container runtime, so
it's better to clean up as much as possible and make sure that Cilium
agent is going to clean up everything after start.

Fixes: #8313

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8335)
<!-- Reviewable:end -->

vadorovsky committed:
cni: Avoid returning error in DEL command

CNI DEL operations might be never retried by the container runtime, so
it's better to clean up as much as possible and make sure that Cilium
agent is going to clean up everything after start.

Fixes: #8313

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

vadorovsky commented on plugins/cilium-cni/cilium-cni.go:
```diff
@@ -544,7 +544,9 @@ func cmdDel(args *skel.CmdArgs) error {
 		)
 
 		if chainAction.ImplementsDelete() {
-			return chainAction.Delete(context.TODO(), ctx)
```
@tgraf Why is CNI plugin returning here immediately? We don't want to perform cleanups below (like removing an endpoint) after running chained plugin(s)? Is there something which I'm missing?

I'm wondering whether I should `return` here or not. So far I removed the `return` and I'm just logging an error.

tgraf replied:
If the chaining implements delete then it will release all resources in that delete command.

tgraf commented on plugins/cilium-cni/cilium-cni.go:
```diff
@@ -529,7 +529,7 @@ func cmdDel(args *skel.CmdArgs) error {
 	if err != nil {
 		// this error can be recovered from
-		return fmt.Errorf("unable to connect to Cilium daemon: %s", err)
+		logger.WithError(err).Warningf("unable to connect to Cilium daemon")
```
I don't think we can continue here as `c` will be nil.

tgraf commented on plugins/cilium-cni/cilium-cni.go:
```diff
@@ -562,11 +564,6 @@ func cmdDel(args *skel.CmdArgs) error {
 		//              retried
 		log.WithError(err).Warning("Errors encountered while deleting endpoint")
-		if clientError, ok := err.(client.ClientError); ok {
-			if clientError.Recoverable() {
```
Is something still using `Recoverable()`?

vadorovsky replied:
Yes, the chained veth plugin is using it.

https://github.com/cilium/cilium/blob/master/plugins/cilium-cni/chaining/generic-veth/generic-veth.go#L215

Should I remove it there as well? Maybe we should avoid returning errors in chained plugins as well.

vadorovsky replied:
For now I removed it. Let me know if you think that it should remain.

vadorovsky committed:
client: Remove ClientError

ClientError existed only to provide an information whether the error is
recoverable or not. CNI plugin is not using that information anymore, so
we can just remove the whole struct and use stdlib for creating client
errors.

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>
